frontend/pkg/ocm/ocm.go

```diff
@@ -0,0 +1,148 @@
+package ocm
+
+import (
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	v1 "github.com/openshift/api/config/v1"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+const (
+	csCloudProvider    string = "azure"
+	csProductId        string = "aro"
+	csHypershifEnabled bool   = true
+	csMultiAzEnabled   bool   = true
+	csCCSEnabled       bool   = true
+)
+
+// ConvertCStoFrontend converts a CS Cluster object into HCPOpenShiftCluster object
+func ConvertCStoFrontend(cluster *cmv1.Cluster) (*api.HCPOpenShiftCluster, error) {
+	hcpcluster := &api.HCPOpenShiftCluster{
+		TrackedResource: arm.TrackedResource{
+			Location: "",
+			Tags:     nil,
+			Resource: arm.Resource{
+				ID:   cluster.ID(),
+				Name: cluster.Name(),
+				Type: cluster.Flavour().ID(),
+				SystemData: &arm.SystemData{
+					CreatedBy:          "",
+					CreatedByType:      "",
+					CreatedAt:          nil,
+					LastModifiedBy:     "",
+					LastModifiedByType: "",
+					LastModifiedAt:     nil,
+				},
+			},
+		},
+		Properties: api.HCPOpenShiftClusterProperties{
+			Spec: api.ClusterSpec{
+				Version: api.VersionProfile{
+					ID:                cluster.Version().ID(),
+					ChannelGroup:      cluster.Version().ChannelGroup(),
+					AvailableUpgrades: cluster.Version().AvailableUpgrades(),
+				},
+				DNS: api.DNSProfile{
+					BaseDomain:       cluster.DNS().BaseDomain(),
+					BaseDomainPrefix: cluster.DomainPrefix(),
+				},
+				Network: api.NetworkProfile{
+					NetworkType: api.NetworkType(cluster.Network().Type()),
+					PodCIDR:     cluster.Network().PodCIDR(),
+					ServiceCIDR: cluster.Network().ServiceCIDR(),
+					MachineCIDR: cluster.Network().MachineCIDR(),
+					HostPrefix:  int32(cluster.Network().HostPrefix()),
+				},
+				Console: api.ConsoleProfile{
+					URL: cluster.Console().URL(),
+				},
+				API: api.APIProfile{
+					URL:        cluster.API().URL(),
+					IP:         "",
+					Visibility: api.Visibility(cluster.API().Listening()),
+				},
+				FIPS:                          cluster.FIPS(),
+				EtcdEncryption:                cluster.EtcdEncryption(),
+				DisableUserWorkloadMonitoring: cluster.DisableUserWorkloadMonitoring(),
+				Proxy: api.ProxyProfile{
+					HTTPProxy:  cluster.Proxy().HTTPProxy(),
+					HTTPSProxy: cluster.Proxy().HTTPSProxy(),
+					NoProxy:    cluster.Proxy().NoProxy(),
+					TrustedCA:  cluster.AdditionalTrustBundle(),
+				},
+				Platform: api.PlatformProfile{
+					ManagedResourceGroup:   cluster.Azure().ManagedResourceGroupName(),
+					SubnetID:               cluster.Azure().SubnetResourceID(),
+					OutboundType:           "",
+					NetworkSecurityGroupID: cluster.Azure().NetworkSecurityGroupResourceID(),
+					EtcdEncryptionSetID:    "",
+				},
+				IssuerURL: "",
+				ExternalAuth: api.ExternalAuthConfigProfile{
+					Enabled:       false,
+					ExternalAuths: []*v1.OIDCProvider{},
```

Comment:
Was `ExternalAuth` intentionally skipped?  I think it's fine to skip it on the 1st pass because, honestly, it's a pain.  But we should leave a TO-DO comment or something similar here, and maybe even add a JIRA story to the backlog so it's not forgotten.

Reply:
Honestly some of these were skipped because they are a bunch of work and we don't have any values just yet so i left things out. 100% we need to track updating these to ensure all objects are set. I can add a comment and create a card if that works for you